docs/guides/deploy-k8s.md

```diff
@@ -18,6 +18,17 @@ This guide explains how to deploy the Sensitive Data Archive (SDA) in kubernetes
 
 - Differences in deployment make concrete examples challenges, explain what can be exemplified and what not in this guide
 
+
+For secure deployment of the system one can think it by what can be accessed from where, for all ways of deploying two trust boundaries can be used, external and internal. For an extra layer of security also the storage trust boundary can be separate. The service is provided for customers on the internet therefore an example of deploying the service is using two separate Kubernetes clusters, one for responding customers and other communication outside, and the other cluster is more secure storage facing internal cluster. One thing to consider is where to release the data, that could be closed protected environment with tightly restricted access, Data out can be put in internal cluster.
+
+The services could be divided into two trust boundaries
+- The services in external in external cluster are Inbox and MQ
+- The services in internal cluster are  Intercept, Ingest, Verify, Mapper, Finalize, Backup and Data out.
+
```

Comment:
Should "Data out" be in the internal cluster?

Reply:
In order the "Data out" to access the data archive it should be in the internal cluster, if the "Data out" is used from trusted environment then it placing it to internal should be ok. 

If the "Data out" is accessed from the internet, that is an another situation, if this is valid case, then an other recommendation perhaps? Not sure if the existing proxy alternative address this?

Reply:
the reasoning for Data Out being in internal cluster is that it might have access to the private key. probably we can have another line mentioning that if the Data Out servers unencrypted files we recommend to have it available only in a internal facing cluster

Reply:
Ok, but does "Data out" have access to data?
How does customer have access to "Data out"? Is internet an option to access "Data out"?
From where is private key fetched?

Fix to modify line 22 to contain the recommendation is at the end.

Reply:
> "Data out" have access to data?

yes, because it provides the stored data

> How does customer have access to "Data out"? Is internet an option to access "Data out"?

currently we use GA4GH passports to control access to data, meaning an issue issues a GA4GH Visa and that get recorded in the passport and if this corresponds with the dataset that is registered then access is allowed. The full specification is at https://github.com/ga4gh-duri/ga4gh-duri.github.io/blob/master/researcher_ids/ga4gh_passport_v1.md#overview and the flow is illustrated at: https://ga4gh.github.io/data-security/aai-openid-connect-profile#full-login-and-token-exchange-interaction

> From where is private key fetched?

simple answer: a path.
that path depends on the secret management that is associated the deployment strategy. it can be retrieved from a secret storage such as (hasicorp vault or https://github.com/bitnami-labs/sealed-secrets) or inject with a k8s sidecar or stored in an encrypted k8s secret or anything equivalent.

Reply:
Thanks for the explanations! If i get it correctly, this is not related to the discussion about having different data out points for direct and indirect access. (there might be the case that internal apps need to access the data for inferance)(and i guess this is another case)